### PR TITLE
Document Dependency management and produceExecutableDuringBuild()/produceFeatureDuringBuild() methods

### PR DESCRIPTION
**Pull request by denisbedin**
https://docs.microej.com/en/feature-microej-sdk-6-develop/

### CODE REVIEW

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
+.. _gradle_dependencies_management_chapter:
+
+Dependencies management
+=======================
+
+This chapter describes all the dependency configurations added to your project by the MicroEJ Gradle plugins.
+
+.. note::
+
+   The MicroEJ Gradle plugins extends the Gradle Java and Java Library plugins.
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:10]:
```suggestion
   The MicroEJ Gradle plugins extend the Gradle Java and Java Library plugins.
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+--------------------
+
+The :ref:`sdk6_module_natures.application` plugin defines a list of variants 
+that are used during the publication of an Application.
+
+microejWPK
+~~~~~~~~~~
+
+This variant is used to publish the WPK of an Application, 
+that can be fetched by declaring a dependency against the ``microejApplication`` configuration.
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:36]:
```suggestion
that can be fetched by declaring a dependency with the ``microejApplication`` configuration.
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+The LibaryElement is set to ``microej-wpk``.
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:37]:
```suggestion
The ``LibraryElement`` attribute of the variant is set to ``microej-wpk``.
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+microejExecutable
+~~~~~~~~~~~~~~~~~
+
+This variant is used to publish the Executable of an Application, 
+that can be fetched by declaring a dependency against the ``microejKernel`` configuration.
+The LibaryElement is set to ``microej-executable``.
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:44]:
```suggestion
The ``LibraryElement`` attribute of the variant is set to ``microej-executable``.
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+This variant is used to publish the WPK of an Application, 
+that can be fetched by declaring a dependency against the ``microejApplication`` configuration.
+The LibaryElement is set to ``microej-wpk``.
+
+microejExecutable
+~~~~~~~~~~~~~~~~~
+
+This variant is used to publish the Executable of an Application, 
+that can be fetched by declaring a dependency against the ``microejKernel`` configuration.
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:43]:
```suggestion
that can be fetched by declaring a dependency with the ``microejKernel`` configuration.
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+The LibaryElement is set to ``microej-executable``.
+
+microejExecutableBuildFiles
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This variant is used to publish the files generated when building the Executable of an Application.
+The LibaryElement is set to ``microej-build-files``.
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:50]:
```suggestion
The ``LibraryElement`` attribute of the variant is set to ``microej-build-files``.
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+microejVirtualDevice
+~~~~~~~~~~~~~~~~~~~~
+
+This variant is used to publish the Virtual Device of an Application, 
+that can be fetched by declaring a dependency against the ``microejVeePort`` or the ``microejKernel`` configuration.
+The LibaryElement is set to ``microej-vee-port``.
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:57]:
```suggestion
The ``LibraryElement`` attribute of the variant is set to ``microej-vee-port``.
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This variant is used to publish the files generated when building the Executable of an Application.
+The LibaryElement is set to ``microej-build-files``.
+
+microejVirtualDevice
+~~~~~~~~~~~~~~~~~~~~
+
+This variant is used to publish the Virtual Device of an Application, 
+that can be fetched by declaring a dependency against the ``microejVeePort`` or the ``microejKernel`` configuration.
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:56]:
```suggestion
that can be fetched by declaring a dependency with the ``microejVeePort`` or the ``microejKernel`` configuration.
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+The LibaryElement is set to ``microej-vee-port``.
+
+..
+   | Copyright 2008-2023, MicroEJ Corp. Content in this space is free
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:60]:
```suggestion
   | Copyright 2008-2024, MicroEJ Corp. Content in this space is free 
```

File: SDK6UserGuide/appendix/dependenciesManagement.rst [hunk 10]
```diff
@@ -0,0 +1,64 @@
+.. _gradle_dependencies_management_chapter:
+
+Dependencies management
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/appendix/dependenciesManagement.rst:3]: Shouldn't it be "Dependencies Configurations" instead?

File: SDK6UserGuide/publishProject.rst [hunk 11]
```diff
@@ -47,9 +47,21 @@ The following artifacts are automatically published:
 - The Ivy descriptor file (to allow SDK 5 project to fetch it).
 - The WPK file, if the project is an Application.
 - The Virtual Device, if the project is an Application containing a Java main class.
-- The Executable file, if the project is an Application containing a Java main class.
-- The Feature file, if the project is an Application containing a Java class that implements the ``ej.kf.FeatureEntryPoint`` interface.
 
+If the project is an Application containing a Java main class, the Executable can be published by 
+calling the ``produceExecutableDuringBuild()`` method in the ``microej`` configuration block of the Gradle build file of the project::
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/publishProject.rst:52]:
```suggestion
```suggestion
If the project is an Application containing a Java main class, the Executable is not built and published by default (when launching a ``./gradlew build`` or a ``./gradlew publish`` for example).
This default behavior can be changed by adding the ``produceExecutableDuringBuild()`` method in the ``microej`` configuration block of the Gradle build file of the project::
```
```

File: SDK6UserGuide/publishProject.rst [hunk 12]
```diff
@@ -47,9 +47,21 @@ The following artifacts are automatically published:
...
+
+   microej {
+     produceExecutableDuringBuild()
+   }
+
+If the project is an Application containing a Java class that implements the ``ej.kf.FeatureEntryPoint`` interface, 
+the Feature can be published by calling the ``produceFeatureDuringBuild()`` method in the ``microej`` configuration block of the Gradle build file 
+of the project::
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/publishProject.rst:60]: Same change here.

**Review by tdelhomenie-microej — CHANGES_REQUESTED** [review]

File: SDK6UserGuide/appendix/dependenciesManagement.rst (hunk 2 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/dependenciesManagement.rst]: Done.

File: SDK6UserGuide/appendix/dependenciesManagement.rst (hunk 3 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/dependenciesManagement.rst]: Done.

File: SDK6UserGuide/appendix/dependenciesManagement.rst (hunk 4 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/dependenciesManagement.rst]: Done.

File: SDK6UserGuide/appendix/dependenciesManagement.rst (hunk 5 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/dependenciesManagement.rst]: Done.

File: SDK6UserGuide/appendix/dependenciesManagement.rst (hunk 6 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/dependenciesManagement.rst]: Done.

File: SDK6UserGuide/appendix/dependenciesManagement.rst (hunk 7 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/dependenciesManagement.rst]: Done.

File: SDK6UserGuide/appendix/dependenciesManagement.rst (hunk 9 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/dependenciesManagement.rst]: Done.

File: SDK6UserGuide/appendix/dependenciesManagement.rst (hunk 10 above)
Comment (denisbedin) [reply SDK6UserGuide/appendix/dependenciesManagement.rst]: I renamed the chapter.

File: SDK6UserGuide/publishProject.rst (hunk 11 above)
Comment (denisbedin) [reply SDK6UserGuide/publishProject.rst]: Done.

File: SDK6UserGuide/publishProject.rst (hunk 12 above)
Comment (denisbedin) [reply SDK6UserGuide/publishProject.rst]: Done.

**Pull request merged**